test/oauth2_unittest.cc

```diff
@@ -50,26 +63,129 @@ TEST_F(OAuth2Test, GetHeaderValueUsingTokenFromCredentials) {
   ));
   Environment environment(config);
   OAuth2 auth(environment);
-  SetTokenEndpointForTest(&auth, server.GetUrl() + "/oauth2/v3/token");
+  SetTokenEndpointForTest(&auth, oauth_server.GetUrl() + "/oauth2/v3/token");
 
   EXPECT_EQ("Bearer the-access-token", auth.GetAuthHeaderValue());
-}
 
-TEST_F(OAuth2Test, GetHeaderValueUsingTokenFromMetadataServer) {
-  testing::FakeServer server;
-  server.SetResponse("/instance/service-accounts/default/token",
-                     "{\"access_token\": \"the-access-token\","
-                     " \"token_type\": \"Bearer\","
-                     " \"expires_in\": 3600}");
+  // Verify the POST contents sent to the token endpoint.
+  EXPECT_EQ(1, post_count);
+  EXPECT_EQ("application/x-www-form-urlencoded",
+            post_headers.at("Content-Type"));
```

Comment:
Might want to first check that the key is in the map... Unless you want your tests to throw `std::out_of_range` instead of failing if the header is not present.

Reply:
Done.